ably/ably-chat-js
# react-hooks: Add useRoomReactions hook

### Context

* [CHA-385]
* As part of the react hooks development, a new hook to interact with the `RoomReactions` instance of a room was required.

### Description

* Added `useRoomReactions` hook to provide access to the 'RoomReactions` instance. This allows the user to interact with the underlying `RoomReactions` instance, send a room reaction and subscribe to room reaction events.
* Updated the demo app to use the new hook for room reaction handling and removed the old custom hook.
* Updated the readme to document hook usage.
* Added unit tests for the new hook.

### Checklist


### Testing Instructions (Optional)

1. Spin up a local instance of the demo app.
2. Ensure you can send reactions.
3. In another window, ensure you receive the reactions.
